lib/libc/sys/ioctl.c

```diff
@@ -88,9 +92,18 @@ _ioctl(int fd, unsigned long com, ...)
 		} else {
 			data = va_arg(ap, void *);
 		}
-		va_end(ap);
 	} else {
-		data = NULL;
+#ifdef __CHERI_PURE_CAPABILITY__
```

Comment:
Since RISC-V doesn't use an array for varargs we need this to be something like
```suggestion
#if defined(__CHERI_PURE_CAPABILITY__) && defined(__mips__)
```
We should probably have a comment explaining exactly what this is for.

Reply:
Done it!